File: q1758/q1758.java
class Solution {
    public int minOperations(String s) {
        int ops = 0;

        char prevChar = s.charAt(0);
        for (int i = 1; i < s.length(); i++) {
            char currChar = s.charAt(i);
            if (prevChar == currChar) {
                ops++;
                prevChar = prevChar == '1' ? '0' : '1';
            } else {
                prevChar = currChar;
            }
        }
        return Math.min(ops, s.length() - ops);    
    }
}

File: q1758/q1758.py
class Solution:
    def minOperations(self, s: str) -> int:
        ops = 0
        prevChar = s[0]
        for i in range(1, len(s)):
            currChar = s[i]
            if prevChar == currChar:
                ops += 1
                prevChar = '1' if prevChar == '0' else '0'
            else:
                prevChar = currChar
        return min(ops, len(s) - ops)
        